assets/app/features/upgrade/controllers/upgrade-landing.controller.spec.js

```diff
@@ -79,6 +211,69 @@ describe('Upgrade Landing Controller', function() {
         should.exist(controller);
     });
 
+    describe('when created while prepare is not running', function() {
+        it('should not start polling for status', function() {
+            expect(upgradeStatusFactory.waitForStepToEnd).not.toHaveBeenCalled();
+        });
+
+        it('prepare should not be running', function() {
+            assert.isFalse(controller.prepare.running);
+        });
+
+        it('prepare should not be completed', function() {
+            assert.isFalse(controller.prepare.completed);
+        });
+    });
+
+    describe('when created while prepare is running', function() {
+        beforeEach(function() {
+            // local change in mocked service
+            upgradeFactory.getStatus =
+                sinon.spy(function () { return $q.when(incompleteUpgradeResponse); });
```

Comment:
My implementation is direct extract of what bardJS is doing inside `mockService`. Using jasmine here would cause the service mock to have inconsistent implementation. Some functions would be mocked using sinon and some using jasmine. Not sure if the interfaces are 100% complatible. If not, then using those spies in expectations would require different syntax depending on the library used for particular function.

Anyway, I was thinking about a different approach. Why don't we create an utility library for our tests (e.g. SUSE Test Utils = stu)? The call would then be something like: `stu.remockService(upgradeFactory, { getStatus: $q.when(incompleteUpgradeResponse) } )`. This would mimic the `mockService()` syntax from bardJS and could be reused in other places.